src/structures/BaseGuildVoiceChannel.js

```diff
@@ -20,20 +47,46 @@ class BaseGuildVoiceChannel extends GuildChannel {
       this.rtcRegion = data.rtc_region;
     }
 
-    if ('bitrate' in data) {
+    if ('user_limit' in data) {
       /**
-       * The bitrate of this voice-based channel
+       * The maximum amount of users allowed in this channel.
        * @type {number}
        */
-      this.bitrate = data.bitrate;
+      this.userLimit = data.user_limit;
     }
 
-    if ('user_limit' in data) {
+    if ('video_quality_mode' in data) {
       /**
-       * The maximum amount of users allowed in this channel.
+       * The camera video quality mode of the channel.
+       * @type {?VideoQualityMode}
+       */
+      this.videoQualityMode = VideoQualityModes[data.video_quality_mode];
+    } else {
+      this.videoQualityMode ??= null;
+    }
+
+    if ('last_message_id' in data) {
+      /**
+       * The last message id sent in the channel, if one was sent
+       * @type {?Snowflake}
+       */
+      this.lastMessageId = data.last_message_id;
+    }
+
+    if ('messages' in data) {
+      for (const message of data.messages) this.messages._add(message);
+    }
+
+    if ('rate_limit_per_user' in data) {
+      /**
+       * The rate limit per user (slowmode) for this channel in seconds
        * @type {number}
        */
-      this.userLimit = data.user_limit;
+      this.rateLimitPerUser = data.rate_limit_per_user;
+    }
+
+    if ('nsfw' in data) {
+      this.nsfw = Boolean(data.nsfw);
```

Comment:
```suggestion
      this.nsfw = data.nsfw;
```
is it required to use the Boolean conversion here? the API would send this as a boolean if it was included

Reply:
1. Old code was the same
2. It can be null, so we just narrow it to false

Reply:
according to the [documentation](https://discord.com/developers/docs/resources/channel#channel-object-channel-structure) if `nsfw` is present then it will be a `boolean` and not null. You are right that when the channel is initialized it may not be a boolean. It may be worth a new PR for fixing this if this is indeed the case!

Reply:
ups yeah